### DISCUSSION TITLE
Is it possible to identify a dead position using Stockfish

### DISCUSSION THREAD

**Discussion by dlbbld** [official-stockfish/Stockfish#3914] [Q&A]
The below two positions at depth 20 both have a score of 0.

8/8/3k4/p1p1p1p1/P1P1P1P1/3K4/8/8 w - - 0 1
q3nr1k/4bppp/3p4/rB2nPP1/3BP2P/Np2Q3/1P6/1K1R3R w - - 7 26

Obviously, the first position is a dead position and the second one is not. Is it possible from Stockfish analysis to deduce that Stockfish found a dead position?

**Comment by pb00067** [comment]
When SF shows a score of 0 and the shown hashsize for the transposition table does not increase anymore although it isn't full yet,
you can assume that all analyzed positions read to a draw (by repetition) so that no more positions can be explored.
In such cases you can assume pretty sure that the position is a dead draw.

**Comment by ghost** [comment]
> The below two positions at depth 20 both have a score of 0.

you can search position with multithreading to find better answer , perhaps score was not 0 

`setoption name threads value 32`

**Comment by vondele** [comment]
@movepicker please stop posting these answers, they are not accurate.

**Reply by ghost** [reply]
yes I know 
I reply only some of their question because with multi threading maybe score was not 0 & different scores , I will pay more attention in the future

**Comment by peregrineshahin** [comment]
I really have come to conclusions that this movepicker guy is a smart child who likes chess and code, no way he's wise enough not to spam.

**Comment by dlbbld** [comment]
I have not phrased my question precisely, sorry.

What I would like to know is the following. For the below position, which is a dead position, Stockfish gives a score of 0.0.
8/8/3k4/p1p1p1p1/P1P1P1P1/3K4/8/8 w - - 0 1

Has Stockfish an algorithm that detects that this position is a dead position and therefore gives a score of 0.0?

Or is the score based on evaluating lines, where Stockfish sees no player has an advantage?

**Reply by Disservin** [reply]
That position is very equal so stockfish can easily detect draw and thus gives 0.0 if you take the same position and add two light (or one) coloured bishops for white my lichess stockfish gives +2 to +4 eval even though the position is drawn because of the same reasons. White cannot make any progress. Also now with nnue we dont know if it understands these positions.. there might be an hce terms for dead positions but overall its just, no advantage.

**Reply by dlbbld** [reply]
The question is if Stockfish has dead position detection. With one white light-squared bishop added, Stockfish sees White better in the position. With additionally one black black-squared bishop, evaluation is again 0.0. So it looks like Stockfish is guessing, not seeing that the position is dead. But well that is my question, I can't answer myself.

By the way, the CHA tool recognizes the previous positions as dead positions in no time. It has a way to do quick checks without searching the variation tree. That is the reason it is so fast. See discussion #3916.
